apps/ledger-live-desktop/tests/specs/services/dapp.spec.ts

```diff
@@ -6,110 +6,198 @@ import { Drawer } from "../../page/drawer/drawer";
+    const layout = new Layout(page);
+    const deviceAction = new DeviceAction(page);
 
-test("Wallet API methods @smoke", async ({ page, electronApp }) => {
-  const discoverPage = new DiscoverPage(page);
-  const drawer = new Drawer(page);
-  const modal = new Modal(page);
-  const layout = new Layout(page);
-  const deviceAction = new DeviceAction(page);
+    await layout.goToDiscover();
+    await discoverPage.openTestApp();
+    await drawer.continue();
+    await drawer.waitForDrawerToDisappear();
 
-  await layout.goToDiscover();
-  await discoverPage.openTestApp();
-  await drawer.continue();
-  await drawer.waitForDrawerToDisappear();
+    const [, webview] = electronApp.windows();
 
-  const [, webview] = electronApp.windows();
+    // Checks that we support EIP 6963
+    await webview.click("#provider > button");
 
-  // Checks that we support EIP 6963
-  await webview.click("#provider > button");
+    webview.getByText("Name: Ledger Live");
+    webview.getByText("Network: 1");
+    webview.getByText("ChainId: 0x1");
+    webview.getByText("Accounts: 0x6EB963EFD0FEF7A4CFAB6CE6F1421C3279D11707");
 
-  webview.getByText("Name: Ledger Live");
-  webview.getByText("Network: 1");
-  webview.getByText("ChainId: 0x1");
-  webview.getByText("Accounts: 0x6EB963EFD0FEF7A4CFAB6CE6F1421C3279D11707");
+    // Checks that getAccounts returns the correct account
+    await webview.click("#getAccounts");
 
-  // Checks that getAccounts returns the correct account
-  await webview.click("#getAccounts");
+    webview.getByText("eth_accounts result: 0x6EB963EFD0FEF7A4CFAB6CE6F1421C3279D11707");
 
-  webview.getByText("eth_accounts result: 0x6EB963EFD0FEF7A4CFAB6CE6F1421C3279D11707");
+    // Checks that personalSign works
+    await webview.click("#personalSign");
+    await expect(page.getByText("Sign message")).toBeVisible();
+    await modal.continueToSignTransaction();
 
-  // Checks that personalSign works
-  await webview.click("#personalSign");
-  await expect(page.getByText("Sign message")).toBeVisible();
-  await modal.continueToSignTransaction();
+    // Step Device
+    await deviceAction.silentSign();
+    // Needs a second sign to close the sign message modal (to fix later)
+    await deviceAction.silentSign();
 
-  // Step Device
-  await deviceAction.silentSign();
-  // Needs a second sign to close the sign message modal (to fix later)
-  await deviceAction.silentSign();
+    // Doesn't seem to wait correctly for the modal to disappear as we can still comment the line above and works
+    await modal.waitForModalToDisappear();
+
+    // Improve the deviceAction mocking to return a result in the webview to test
+    // await webview.getByText("Result:");
+
+    // You can uncomment this to make sure visually that it ends correctly
+    // await page.waitForTimeout(10000);
+  });
+});
+
+test.describe("1inch dapp", () => {
+  test.beforeAll(async () => {
+    process.env.MOCK_REMOTE_LIVE_MANIFEST = JSON.stringify([
```

Comment:
would it make more sense to move this JSON to a separate file to make this class more readable?